### ISSUE TITLE
Writing files with '.' in the simTitle

### ISSUE THREAD

**Issue by gundersona** [frkasper/MacroUtils#19]
I require writing my sim files and pictures with a name containing doubles (for example, ud.simTitle = height_6.19_inches). The problem I have is that several of the methods in Write.java call the method 'fileBasename()' which is defined in 'GetStrings.java'. This splits the file name by the '.' and cuts out the second half. From my example above, if I used mu.io.write.picture(sceneName, ud.simTitle, 1300, 800, vo) to write a picture, it writes out 'height_6.png' and cuts out the '.19_inches' from my simTitle. 

I have been working around this by commenting out 
else {
            name = _get.strings.fileBasename(name);
        }
in lines 52-54 of Write.java.

Would you consider providing a fix for this in the source code so that with each new version of starccm+ I can simply use your pre-compiled MacroUtils.jar? 

Thanks!

**Comment by frkasper** [comment]
Hi @gundersona,

Thanks for the feedback. Would something like this work for you?

`ud.sanitizeFileNames = false;`

Then all you need is to include this call after you initialize MacroUtils. That should take care of this symptom everywhere while keeping standard jar file as is. Still need to investigate further though.

Cheers,
Fabio

**Comment by gundersona** [comment]
Yes, something like that would work well I think and sounds easy to implement. 

Thanks,
Andrew

**Comment by frkasper** [comment]
Hi again,

I guess the issue is in here:
```
    private boolean _tryWritePic(NamedObject no, String name, int resx, int resy, boolean vo) {
        if (name == null) {
            name = no.getPresentationName();
        } else {
            name = _get.strings.fileBasename(name);
        }
```

The intention of `_get.strings.fileBasename(name)` is for file names only. Deinitely a bug then. 

The good news is that solution may be even simpler, i.e.: 
```
    private boolean _tryWritePic(NamedObject no, String name, int resx, int resy, boolean vo) {
        if (name == null) {
            name = no.getPresentationName();
        } 
```

Would you please try at your end and tell me if this satisfies your request?

Many thanks!
Fabio

**Comment by frkasper** [comment]
I'm working on this one. Should be fixed shortly.

**Comment by frkasper** [comment]
Fixed in v13.04 branch. Please create a new bug if this is still causing trouble.

```
[*] 
[*] +------------------------------------------------------------------------------------------------------------------------
[*] + 05/25/2018 11:52:42
[*] + Writing a Picture
[*] +------------------------------------------------------------------------------------------------------------------------
[*] Scene name: "Block 3.4 x 4.0 x 7.0 mm^3".
[*] Written: "Block_3.4_x_4.0_x_7.0_mm^3.png".
[*] OK!

[*] 
[*] +------------------------------------------------------------------------------------------------------------------------
[*] + 05/25/2018 11:52:43
[*] + Saving Simulation File
[*] +------------------------------------------------------------------------------------------------------------------------
Saving: Block_95.2_mm^3.sim
Object state saved to Block_95.2_mm^3.sim (212.82KB in 0.023803s).
[*] OK!
```

Thanks again for using MacroUtils.
Fabio
